### PR TITLE
Cleanup emcmake/emconfigure

### PR DESCRIPTION
**Pull request by sbc100**
- Set CMAKE_CROSSCOMPILING_EMULATOR in the same way we set
  CMAKE_TOOLCHAIN_FILE.
- Consistent echoing of make/configure command.
- Don't silently succeed if passed no args

### CODE REVIEW

File: tools/shared.py [hunk 1]
```diff
@@ -1700,64 +1701,62 @@ def remove_sh_exe_from_path(env):
...
+      # compilation with emcc, but instead do builds natively with Clang. This
+      # is a heuristic emulation that may or may not work.
       env['EMMAKEN_JUST_CONFIGURE'] = '1'
-    if EM_BUILD_VERBOSE >= 3:
-      print('configure: ' + str(args), file=sys.stderr)
     if EM_BUILD_VERBOSE >= 2:
       stdout = None
     if EM_BUILD_VERBOSE >= 1:
       stderr = None
+    print('configure: ' + ' '.join(args), file=sys.stderr)
```
Comment (kripken) [inline tools/shared.py]: is this extra logging intended? this seems quite verbose by default

File: tools/shared.py [hunk 2]
```diff
@@ -1700,64 +1701,62 @@ def remove_sh_exe_from_path(env):
...
     run_process(args, stdout=stdout, stderr=stderr, env=env, **kwargs)
-    if 'EMMAKEN_JUST_CONFIGURE' in env:
-      del env['EMMAKEN_JUST_CONFIGURE']
```
Comment (kripken) [inline tools/shared.py]: why don't we need to clean this up?

File: tools/shared.py (hunk 1 above)
Comment (sbc100) [reply tools/shared.py]: Its for consistency with the "make:" logging below.  I can't turn them both into `logging.debug` if you prefer, but i think its useful to always see the underlying command that it runs.

File: tools/shared.py (hunk 2 above)
Comment (sbc100) [reply tools/shared.py]: Because env is local thing.. its does live after this function.

Comment (kripken) [reply tools/shared.py]: but env can be passed in as a parameter, in which case we modify it? (do we copy it somewhere that I missed?)

File: tools/shared.py (hunk 1 above)
Comment (kripken) [reply tools/shared.py]: if it's more consistent that seems fine then.

File: tools/shared.py (hunk 2 above)
Comment (sbc100) [reply tools/shared.py]: Oh I think i had an env.copy somewhere in a previous version of this change.   I'll re-add that.

Comment (kripken) [reply tools/shared.py]: lgtm with that

**Pull request merged**